crates/oxc_hir/src/hir_util.rs

```diff
@@ -371,3 +377,65 @@ pub fn get_boolean_value(expr: &Expression) -> Option<bool> {
         _ => None,
     }
 }
+
+/// Port from [closure-compiler](https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L234)
+/// Gets the value of a node as a String, or `None` if it cannot be converted. When it returns a
+/// String, this method effectively emulates the `String()` JavaScript cast function.
+/// This method does not consider whether `expr` may have side effects.
+fn get_string_value<'a>(expr: &'a Expression) -> Option<Cow<'a, str>> {
+    match expr {
+        Expression::StringLiteral(string_literal) => Some(string_literal.value.as_str().into()),
+        Expression::TemplateLiteral(template_literal) => {
+            // TODO: I don't know how to iterate children of TemplateLiteral in order,so only checkout string like `hi`.
+            // Closure-compiler do more: [case TEMPLATELIT](https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L241-L256).
+            if let Some(quasi) = template_literal.quasis.get(0) && quasi.tail {
+                quasi.value.cooked.as_ref().map(|cooked| cooked.as_str().into())
+            } else {
+                None
+            }
+        }
+        Expression::Identifier(ident) => {
+            let name = ident.name.as_str();
+            if matches!(name, "undefined" | "Infinity" | "NaN") { Some(name.into()) } else { None }
+        }
+        Expression::NumberLiteral(number_literal) => Some(number_literal.value.to_string().into()),
+        Expression::BigintLiteral(big_int_literal) => {
+            Some(format!("{}n", big_int_literal.value).into())
+        }
+        Expression::NullLiteral(_) => Some("null".to_string().into()),
+        Expression::BooleanLiteral(bool_literal) => Some(bool_literal.value.to_string().into()),
+        Expression::UnaryExpression(unary_expr) => {
+            match unary_expr.operator {
+                UnaryOperator::Void => Some("undefined".to_string().into()),
+                UnaryOperator::LogicalNot => {
+                    // need reversed.
+                    get_boolean_value(&unary_expr.argument)
+                        .map(|boolean| (!boolean).to_string().into())
+                }
+                _ => None,
+            }
+        }
+        Expression::ArrayExpression(_) => {
+            // TODO: https://github.com/google/closure-compiler/blob/e13f5cd0a5d3d35f2db1e6c03fdf67ef02946009/src/com/google/javascript/jscomp/NodeUtil.java#L302-L303
+            None
+        }
+        Expression::ObjectExpression(_) => Some("[object Object]".to_string().into()),
```

Comment:
Does Cow::owned or Cow::borrowed work? This is static so it has a longer lifetime